Extend observability to connect to external cos

hemanthnakkina opened the pull request:
New Observability commands:
* Enable Observability with external COS:
`sunbeam enable observability external `

* Disable Observability with external COS:
`sunbeam disable observability external`

* Enable Observability with internal:
`sunbeam enable observability embedded`

* Disable Observability with internal:
`sunbeam disable observability embedded`

Backward compatible commands that are deprecated:
* Enable Observability with internal
`sunbeam enable observability`

* Disable Observability with internal
`sunbeam disable observability`

gboutry commented:
I wonder if it does not make more sense to have this separated as another plugin ?

Make them work like TLS, a Mutual Exclusive plugin.

Thinking about the UX, having to either have all the options or none of them does not make much sense in a CLI pov, but having:

`sunbeam enable observability`

`sunbeam enable observability-ext offer1 offer2 offer3` (notice no params, only mandatory arguments)


We can also discuss who gets the `observability` name. I think the external one should get it, as it is the one we would want people to use the most.

Maybe `observability-embedded`? Longer to type, and shows that it's deployed inside the cluster?

hemanthnakkina commented:
> I wonder if it does not make more sense to have this separated as another plugin ?
> 
> Make them work like TLS, a Mutual Exclusive plugin.
> 
> Thinking about the UX, having to either have all the options or none of them does not make much sense in a CLI pov, but having:
> 
> `sunbeam enable observability`
> 
> `sunbeam enable observability-ext offer1 offer2 offer3` (notice no params, only mandatory arguments)
> 
> We can also discuss who gets the `observability` name. I think the external one should get it, as it is the one we would want people to use the most.
> 
> Maybe `observability-embedded`? Longer to type, and shows that it's deployed inside the cluster?

Modified as per the dicussion and updated the PR commit message to list the new commands.
